### PR TITLE
Continue execution if there are nosem mismatches with --strict

### PR DESCRIPTION
**Pull request by minusworld**
--strict will will report a nonzero exit code if there are nosem mismatches.

PR checklist:
- [x] changelog is up to date

### CODE REVIEW

**Comment by minusworld** [comment]
Fixes https://github.com/returntocorp/semgrep/issues/3099

**Comment by minusworld** [comment]
Also fixes https://github.com/returntocorp/enterprise/issues/614

File: semgrep/semgrep/semgrep_main.py [hunk 1]
```diff
@@ -260,13 +261,16 @@ def main(
...
+    nosem_errors = []
+    for rule, rule_matches in rule_matches_by_rule.items():
+        evolved_rule_matches = []
+        for rule_match in rule_matches:
+            ignored, returned_errors = rule_match_nosem(rule_match, strict)
+            evolved_rule_matches.append(attr.evolve(rule_match, is_ignored=ignored))
+            nosem_errors.extend(returned_errors)
+        rule_matches_by_rule[rule] = evolved_rule_matches
+
+    output_handler.handle_semgrep_errors(nosem_errors)
```
Comment (mschwager) [inline semgrep/semgrep/semgrep_main.py:273]: Should we instead append to `semgrep_errors` and make one call to `handle_semgrep_errors`? Not sure if that matters in our `output_handler` error handling.

Comment (brendongo) [reply semgrep/semgrep/semgrep_main.py:273]: Doesn't matter functionality-wise. Seems ok to add it to its own list and call handle_semgrep_errors twice

**Review by brendongo — APPROVED** [review]

**Pull request merged**